designs/disruption-controls-by-reason.md

```diff
@@ -0,0 +1,312 @@
+# Disruption Controls By Reason
+## TOC and Overview
+1. [Overview](#overview)
+2. [User Scenarios](#user-scenarios)
+3. [Known Requirements](#clarifying-the-requirements-and-behavior)
+   - [Q: Default or Undefined Reason Case Handling](#q-default-or-undefined-reason-case-handling)
+   - [Q: Calculation of Allowed Disruptions in a Multi-Reason World](#q-calculation-of-allowed-disruptions-in-a-multi-reason-world)
+   - [Q: Handling an Unspecified Default Reason](#q-handling-an-unspecified-default-reason)
+   - [Q: Definition of Default in Budgets](#q-definition-of-default-in-budgets)
+4. [API Design](#api-design)
+   - [Approach A: Extending the Budget API to Specify a Reason](#approach-a-extending-the-budget-api-to-specify-a-reason)
+      * [List Approach: Multiple Reasons per Budget](#list-approach-multiple-reasons-per-budget)
+      * [Single Reason Approach: One Reason per Budget](#single-reason-approach-one-reason-per-budget)
+      * [Pros and Cons for Both Approaches](#pros-and-cons-for-both-approaches)
+      * [Preferred Option: List Approach](#preferred-option-list-approach)
+   - [Approach B: Defining Per Reason Controls](#approach-b-defining-per-reason-controls)
+      * [Pros and Cons](#pros-and-cons)
+   - [API Design Conclusion: Preferred Design](#api-design-conclusion-preferred-design)
+
+## User Scenarios 
+1. Users need the capability to schedule upgrades only during business hours or within more restricted time windows. Additionally, they require a system that doesn't compromise the cost savings from consolidation when upgrades are blocked due to drift.
+2. Users want to minimize workload disruptions during business hours but still want to be able to delete empty nodes throughout the day.  That is, empty can run all day, while limiting cost savings and upgrades due to drift to non-business hours only.
+
+See Less Made Up Scenarios here: 
+- https://github.com/kubernetes-sigs/karpenter/issues/924 
```

Comment:
This is great! More of our design docs should reference out to issues like this; great model for others!